packages/core/content-type-builder/admin/src/components/FormModal/utils/getCustomFieldDefaultOptions.js

```diff
@@ -0,0 +1,16 @@
+const getCustomFieldDefaultOptions = (options, optionDefaults = []) => {
```

Comment:
Would you be able to make the function more pure/functional? I'm not a fan of mutating the parameter, especially with the recursion

Reply:
Sure, good point, I think I can do that by handling the recursion with Array.reduce